dbms/src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -2568,6 +2568,39 @@ MergeTreeData::DataPartsVector MergeTreeData::getAllDataPartsVector(MergeTreeDat
     return res;
 }
 
+std::vector<DetachedPartInfo>
+MergeTreeData::getDetachedParts() const
+{
+    std::vector<DetachedPartInfo> res;
+
+    for (Poco::DirectoryIterator it(full_path + "detached");
+        it != Poco::DirectoryIterator(); ++it)
+    {
+        res.resize(res.size() + 1);
+        auto & part = res.back();
```

Comment:
BTW, it's pre-C++11 pattern. Now it's better to create an object and move it into vector (if the move is not expensive).

Reply:
I'll use emplace_back if you don't mind, not to write push_back in four places.